### DISCUSSION TITLE
Use worklet for `onChangeText`

### DISCUSSION THREAD

**Discussion by pistonsky** [software-mansion/react-native-reanimated#2788] [Q&A]
Hi 👋 

I have a `TextInput` that is used to enter a phone number. It does some formatting logic, but since everything is delayed one frame, it looks bad (not too bad, but not ideal). I know that I can write my own native component and implement `shouldChangeCharactersInRange` there, but I would like to deal with that on js side. I already know how to control what's shown inside of `TextInput` using `text` native prop in combo with `Animated.addWhitelistedNativeProps({ text: true });`, so the only thing left is to be able to provide a worklet for `onChangeText`. Is there a way to do that?

**Comment by Noitidart** [comment]
I don't know how to use reanimated, but if you make the TextInput uncontrolled it shouldn't lag by that frame.
